helix_fhir_client_sdk/fhir_client.py

```diff
@@ -179,6 +179,52 @@ def limit(self, limit: int) -> "FhirClient":
         self._limit = limit
         return self
 
+    @property
+    def access_token(self) -> Optional[str]:
+        # if we have an auth server url but no access token then get an access token
+        if self._auth_server_url and not self._access_token:
+            assert (
+                self._login_token
+            ), "login token must be present if auth_server_url is set"
+            http: Session = self._create_http_session()
+            self._access_token = self.authenticate(
+                http=http,
+                auth_server_url=self._auth_server_url,
+                auth_scopes=self._auth_scopes,
+                login_token=self._login_token,
+            )
```

Comment:
@adhearn I think this is a good refactor to make it simpler!  thanks for the effort ;)